lib/prometheus/alert_buffer_client/client.rb

```diff
@@ -49,14 +49,17 @@ def initialize(options = {})
       # @option options [String] :generation_id Database generation Id.
       # @option options [Integer] :from_index Minimal index of alerts to fetch.
       #
+      # @return [Hash] response with keys: generationID, messages
       # All alerts will be fetched if options are omitted.
       def get(options = {})
         response = @client.get do |req|
           req.params['generationID'] = options[:generation_id]
           req.params['fromIndex'] = options[:from_index]
         end
 
-        JSON.parse(response.body)['messages']
+        JSON.parse(response.body)
+      rescue
+        raise RequestError, 'Bad response from server'
```

Comment:
the rescue is for all the method so you return 'Bad response' for other reasons too (e.g no network)

Reply:
you can set up prometheus to return json[1]

[1] https://github.com/moolitayer/manageiq-providers-kubernetes/blob/4cd2b3e6d8f42658664e1d34583d052e92d072ba/app/models/manageiq/providers/kubernetes/prometheus/message_buffer_client.rb#L30

Reply:
@moolitayer do you think I should just propagate the exception raised?

Reply:
That would be a good start. You can keep this in mind for someplace we would need to catch exceptions:
https://github.com/ManageIQ/manageiq-providers-kubernetes/blob/c202228c2b44af89ad2ee90615e45cd6bcf33575/app/models/manageiq/providers/kubernetes/container_manager_mixin.rb#L151

Reply:
@moolitayer does that mean just removing the `rescue` block?

Reply:
Yes